src/cnaas_nms/api/generic.py

```diff
@@ -56,6 +58,51 @@ def offset_results() -> int:
     return offset
 
 
+def pagination_headers(total_count) -> dict:
+    per_page = DEFAULT_PER_PAGE
+    page_arg = 1
+    links = []
+    headers = {
+        "X-Total-Count": total_count,
+    }
+
+    args = request.args
+    if 'per_page' in args:
+        try:
+            per_page_arg = int(args['per_page'])
+            assert 1 <= per_page_arg <= MAX_PER_PAGE
+            per_page = per_page_arg
+        except (AssertionError, ValueError):
+            pass
+
+    last_page = int(math.ceil(total_count / per_page))
+    if last_page == 1:
+        return {}
+
+    if 'page' in args:
+        try:
+            page_arg = max(1, int(args['page']))
+        except ValueError:
+            pass
+
+    query_without_page = {k: request.args[k] for k in request.args if k not in ["page"]}
+
+    if page_arg * per_page < total_count:
+        links.append('<{}>; rel="next"'.format(
+            request.base_url + "?" + urllib.parse.urlencode({**query_without_page, "page": page_arg + 1})
+        ))
+
+    if page_arg != last_page:
```

Comment:
Would we want the "last" link even if `page_arg == last_page`? Why not?

Reply:
I modeled after github API and it seemed like they didn't have the header on the last page, maybe it's easier for client code to not have to parse anything at all for next pages if the header is not there at all

Reply:
```
[johanmarcusson@indy-x1 ~]$ curl -sI "https://api.github.com/search/code?q=addClass+user:mozilla&page=33" | grep link
link: <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=32>; rel="prev", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=34>; rel="next", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=34>; rel="last", <https://api.github.com/search/code?q=addClass+user%3Amozilla&page=1>; rel="first"
[johanmarcusson@indy-x1 ~]$ curl -sI "https://api.github.com/search/code?q=addClass+user:mozilla&page=34" | grep link
[johanmarcusson@indy-x1 ~]$ 
```

Reply:
Ok, makes sense to follow the github implementation 👍